generator.go

```diff
@@ -62,7 +64,15 @@ type Config struct {
 //
 // Members that are definitions, hidden fields, or optional fields are ignored.
 func Generate(val cue.Value, c Config) (b []byte, err error) {
-	if err = val.Validate(); err != nil {
+	file, err := GenerateAST(val, c)
+	if err != nil {
+		return nil, err
+	}
+	return []byte("\n" + file.String()), nil
+}
+
+func GenerateAST(val cue.Value, c Config) (*ts.File, error) {
```

Comment:
One thing i like about having `GenerateAST()` with a return type of `ts.File` is the opportunity it affords the caller to more easily mess around with the final output - e.g., how Grafana has to, for now at least, inject its own import statement(s).

To that end - unless i missed it? - i'd love to see AST types for representing TS imports.